lib/record_store/zone.rb

```diff
@@ -61,8 +61,12 @@ def modified(verbose: false) # rubocop:disable Lint/UnusedMethodArgument
           Thread.new do
             current_zone = nil
             while zones.any?
-              mutex.synchronize { current_zone = zones.shift }
-              mutex.synchronize { modified_zones << current_zone } unless current_zone.unchanged?
+              mutex.synchronize do
+                if zones.any? # recheck zones since there's a race in the while
+                  current_zone = zones.shift
+                  modified_zones << current_zone unless current_zone.unchanged?
+                end
```

Comment:
```suggestion
                next unless zones.any? # recheck zones since there's a race in the while
                current_zone = zones.shift
                modified_zones << current_zone unless current_zone.unchanged?
```